service-base/src/main/java/org/eclipse/hono/service/HonoBeanNameGenerator.java

```diff
@@ -0,0 +1,28 @@
+/**
+ * Copyright (c) 2017 Bosch Software Innovations GmbH.
+ *
+ * All rights reserved. This program and the accompanying materials
+ * are made available under the terms of the Eclipse Public License v1.0
+ * which accompanies this distribution, and is available at
+ * http://www.eclipse.org/legal/epl-v10.html
+ *
+ * Contributors:
+ *    Bosch Software Innovations GmbH - initial creation
+ */
+
+package org.eclipse.hono.service;
+
+import org.springframework.beans.factory.config.BeanDefinition;
+import org.springframework.beans.factory.support.BeanDefinitionRegistry;
+import org.springframework.beans.factory.support.BeanNameGenerator;
+
+/**
+ * Generates bean names that are identical to the class name.
+ */
+public class HonoBeanNameGenerator implements BeanNameGenerator {
```

Comment:
**Background information**

The changes in `AbstractApplication` (`NewAbstractApplication`) impose a changed behaviour by the `ObjectFactory`: It is no longer parametrized with the service type, but with a wildcard (`ObjectFactory<? extends ConfigurationSupportingVerticle>`). In this case Spring's `ObjectFactory` does not use the the full qualified class name, but the generated de-capitalized short name (e.g.: "deviceRegistryAmqpServer"). This means the following configuration does not work any more:
```
    @Bean
    public ObjectFactoryCreatingFactoryBean deviceRegistryServerFactory() {
        ObjectFactoryCreatingFactoryBean factory = new ObjectFactoryCreatingFactoryBean();
        factory.setTargetBeanName(DeviceRegistryAmqpServer.class.getName());
        return factory;
    }
```

I see 3 possible solutions: 

 1. explicitly set Bean name at every component: `@Component("org.eclipse.hono.deviceregistry.DeviceRegistryAmqpServer")` leaving ObjectFactory config as it is
 2. changing the targetBeanname in config for every `ObjectFactoryCreatingFactoryBean` in Hono `ObjectFactoryCreatingFactoryBean factory = new ObjectFactoryCreatingFactoryBean();
        factory.setTargetBeanName("deviceRegistryAmqpServer");` (optionally this could be a constant defined in the component)
 3. using a custom `BeanNameGenerator`:
    ```
    public class HonoBeanNameGenerator implements BeanNameGenerator {
        @Override
        public String generateBeanName(BeanDefinition beanDefinition, BeanDefinitionRegistry beanDefinitionRegistry) {
            return beanDefinition.getBeanClassName();
        }
    }
    ```
    This must be added on every `Application` class like this: 
    ```
    @ComponentScan(basePackages = { "org.eclipse.hono.service", "org.eclipse.hono.deviceregistry" }, nameGenerator = HonoBeanNameGenerator.class)
    @Configuration
    @EnableAutoConfiguration
    public class Application extends AbstractApplication {
    ...
    ```
    This causes all generated bean names to start upper case.